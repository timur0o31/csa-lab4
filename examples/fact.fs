var res 0
: fact
    1 res !
    begin
        dup 0 != if
            dup
            res @ * res !
            dec
        else
            drop
            exit
        then
    again
;
10 fact
halt