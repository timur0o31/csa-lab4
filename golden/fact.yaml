in_source: |
  var res 0
  : fact
      1 res !
      begin
          dup 0 != if
              dup
              res @ * res !
              dec
          else
              drop
              exit
          then
      again
  ;
  10 fact
  halt
in_stdin: |
out_instructions: !!binary |
  /////wQAAAAEAAAKBAAABUgAAABwAAAABAAAAQQAAAAIAAAAXAAAAAQAAAAgAAAABAAAF1gAAABMAAAAXAAAAAQAAAAMAAAAJAAAAAQAAAAIAAAAMAAAAAQAABxEAAAAYAAAAAQAAB0sAAAALAAAAEQAAAB0AAAABAAACEQAAABUAAAA
out_data: !!binary |
  AAAAAA==
out_stdout: |
  source LoC: 16 code instr: 32
  ============================================================

out_instructions_hex: |
  0 - 04000000 - lit 0
  1 - 0400000A - lit 10
  2 - 04000005 - lit 5
  3 - 48000000 - call
  4 - 70000000 - halt
  5 - 04000001 - lit 1
  6 - 04000000 - lit 0
  7 - 08000000 - store
  8 - 5C000000 - dup
  9 - 04000000 - lit 0
  10 - 20000000 - sub
  11 - 04000017 - lit 23
  12 - 58000000 - swap
  13 - 4C000000 - jz
  14 - 5C000000 - dup
  15 - 04000000 - lit 0
  16 - 0C000000 - load
  17 - 24000000 - mul
  18 - 04000000 - lit 0
  19 - 08000000 - store
  20 - 30000000 - dec
  21 - 0400001C - lit 28
  22 - 44000000 - jump
  23 - 60000000 - drop
  24 - 0400001D - lit 29
  25 - 2C000000 - inc
  26 - 2C000000 - inc
  27 - 44000000 - jump
  28 - 74000000 - nop
  29 - 04000008 - lit 8
  30 - 44000000 - jump
  31 - 54000000 - ret
